toolkit/tools/pkg/imagecustomizerlib/configvalidation.go

```diff
@@ -130,30 +135,38 @@ func ValidateConfig(ctx context.Context, baseConfigPath string, config *imagecus
 	return rc, nil
 }
 
-func validateInput(baseConfigPath string, input imagecustomizerapi.Input, inputImageFile string) (string, error) {
-	switch {
-	case inputImageFile != "":
+func validateInput(configChain []*ConfigWithBasePath, inputImageFile string) (string, error) {
+	if inputImageFile != "" {
 		if yes, err := file.IsFile(inputImageFile); err != nil {
 			return "", fmt.Errorf("%w (file='%s'):\n%w", ErrInvalidInputImageFileArg, inputImageFile, err)
 		} else if !yes {
 			return "", fmt.Errorf("%w (file='%s')", ErrInputImageFileNotFile, inputImageFile)
 		}
-
 		return inputImageFile, nil
+	}
 
-	case input.Image.Path != "":
-		inputImageAbsPath := file.GetAbsPathWithBase(baseConfigPath, input.Image.Path)
-		if yes, err := file.IsFile(inputImageAbsPath); err != nil {
-			return "", fmt.Errorf("%w (path='%s'):\n%w", ErrInvalidInputImageFileConfig, input.Image.Path, err)
-		} else if !yes {
-			return "", fmt.Errorf("%w (path='%s')", ErrInputImageFileNotFile, input.Image.Path)
+	// Resolve input image path
+	var inputImageAbsPath string
+	for _, configWithBase := range configChain {
```

Comment:
If you iterate through this list in reverse (`slices.Backward`), then you could early return as soon as you find a non-empty value. This would probably make the code a little cleaner. (The `return "", ErrInputImageFileRequired` would happen at the end of the function after all possible sources have been exhausted.)

You could also do a similar pattern for `outputImageFormat` and `outputImageFile`. (You'll need separate `validateOutputImageFormat()` and `validateOutputImageFile` functions to do this.)